tests/backup/backup_upgrade_test.go

```diff
@@ -759,3 +760,552 @@ var _ = Describe("{PXBackupEndToEndBackupAndRestoreWithUpgrade}", func() {
+				singleNSNamespaces = SafeAppend(&mutex, singleNSNamespaces, namespace).([]string)
+				singleNSScheduleNames = SafeAppend(&mutex, singleNSScheduleNames, singleNSScheduleName).([]string)
+			}
+			_ = TaskHandler(Inst().AppList, createSingleNSBackupTask, Parallel)
+		})
+		var versions []string
+		if len(Inst().SchedUpgradeHops) > 0 {
+			versions = strings.Split(Inst().SchedUpgradeHops, ",")
+		}
+		dash.VerifyFatal(len(versions) > 0, true, "Check if upgrade versions provided is provided")
+
+		for _, version := range versions {
+			Step("Upgrading OCP cluster", func() {
+
+				err := SwitchBothKubeConfigANDContext("source")
+				dash.VerifyFatal(err, nil, "Switching context and kubeconfig to source cluster")
+
+				err = Inst().S.UpgradeScheduler(version)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("verify [%s] upgrade to [%s] is successful", Inst().S.String(), version))
+				PrintK8sCluterInfo()
+
+				err = SwitchBothKubeConfigANDContext("destination")
+				dash.VerifyFatal(err, nil, "Switching context and Kubeconfig to destination cluster")
+
+				err = Inst().S.UpgradeScheduler(version)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("verify [%s] upgrade to [%s] is successful", Inst().S.String(), version))
+				PrintK8sCluterInfo()
+
+				err = SwitchBothKubeConfigANDContext("source")
+				dash.VerifyFatal(err, nil, "Switching context and kubeconfig to source cluster")
+
+			})
+			if Inst().Provisioner == "portworx" {
+				Step("validate storage components", func() {
+
+					err := SwitchBothKubeConfigANDContext("source")
+					dash.VerifyFatal(err, nil, "Switching context and kubeconfig to source cluster")
+
+					urlToParse := fmt.Sprintf("%s/%s", Inst().StorageDriverUpgradeEndpointURL, Inst().StorageDriverUpgradeEndpointVersion)
+					u, err := url.Parse(urlToParse)
+					log.FailOnError(err, fmt.Sprintf("error parsing PX version the url [%s]", urlToParse))
+					err = Inst().V.ValidateDriver(u.String(), true)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("verify volume driver after upgrade to %s", version))
+
+					// Printing cluster node info after the upgrade
+					PrintK8sCluterInfo()
+
+					err = SwitchBothKubeConfigANDContext("destination")
+					dash.VerifyFatal(err, nil, "Switching context and Kubeconfig to destination cluster")
+
+					urlToParse = fmt.Sprintf("%s/%s", Inst().StorageDriverUpgradeEndpointURL, Inst().StorageDriverUpgradeEndpointVersion)
+					u, err = url.Parse(urlToParse)
+					log.FailOnError(err, fmt.Sprintf("error parsing PX version the url [%s]", urlToParse))
+					err = Inst().V.ValidateDriver(u.String(), true)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("verify volume driver after upgrade to %s", version))
+
+					// Printing cluster node info after the upgrade
+					PrintK8sCluterInfo()
+
+					err = SwitchBothKubeConfigANDContext("source")
+					dash.VerifyFatal(err, nil, "Switching context and kubeconfig to source cluster")
+				})
+
+				Step("update node drive endpoints", func() {
```

Comment:
If you see a 'failed to run command on debug pod' error in the previous step, consider swapping this step with the previous one.

Reply:
Done